File: frontend/src/types/job.ts
export type JobStatus = "QUEUED" | "RUNNING" | "SUCCESS" | "FAILED" | "RETRYING" | "CANCELLED";

export interface Job {
  id: string;
  task_type: string;
  payload: Record<string, unknown>;
  status: JobStatus;
  priority: number;
  max_retries: number;
  retry_count: number;
  timeout_seconds: number;
  idempotency_key: string | null;
  result: Record<string, unknown> | null;
  error: string | null;
  celery_task_id: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
}

export interface JobListResponse { items: Job[]; total: number; }

File: backend/app/schemas/job.py
"""Job API schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.job import JobStatus


class JobCreate(BaseModel):
    """Payload used to submit a job."""

    task_type: str = Field(min_length=1, max_length=100)
    payload: dict = Field(default_factory=dict)


class JobResponse(BaseModel):
    """Public representation of a job."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    task_type: str
    payload: dict
    status: JobStatus
    result: dict | None
    error: str | None
    celery_task_id: str | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None


class JobListResponse(BaseModel):
    """Paginated job listing response."""

    items: list[JobResponse]
    total: int
